cmd/git-sync/main.go

```diff
@@ -452,6 +452,14 @@ func addWorktreeAndSwap(ctx context.Context, gitRoot, dest, branch, rev string,
 	}
 	log.V(0).Info("reset worktree to hash", "path", worktreePath, "hash", hash)
 
+	// Update submodules
+
+	_, err = runCommand(ctx, worktreePath, *flGitCmd, "submodule", "update", "--init", "--recursive")
```

Comment:
I thin you should respect the `--depth` flag here, too, shouldn't you?

Reply:
Can you please add a comment here that repos that have no sub-modules will succeed here with no errors?

Reply:
ok, will add test cases soon, I've tested it in our project and it works fine for both cases: repo with submodules and without submodules. But of course it should be covered by tests.

Reply:
> I thin you should respect the `--depth` flag here, too, shouldn't you?

right, how should it be supported? additional command line flag or reuse of existing `--depth` flag ?

Reply:
I've added --depth flag support

Reply:
If we're NOT going to add --depth here, can you leave a TODO explaining why, so we can investigate later?

Reply:
done